src/connection.rs

```diff
@@ -204,32 +213,46 @@ impl ActualConnection {
     }
 
     pub fn send_bytes(&mut self, bytes: &[u8]) -> RedisResult<Value> {
-        let w = match *self {
-            ActualConnection::Tcp(ref mut reader) => reader.get_mut() as &mut Write,
+        match *self {
+            ActualConnection::Tcp(TcpConnection { ref mut reader, .. }) => {
+                try!(reader.get_mut().write_all(bytes));
+                Ok(Value::Okay)
+            }
             #[cfg(any(feature="with-unix-sockets", feature="with-system-unix-sockets"))]
-            ActualConnection::Unix(ref mut sock) => &mut *sock as &mut Write,
-        };
-        try!(w.write_all(bytes));
-        Ok(Value::Okay)
+            ActualConnection::Unix(ref mut connection) => {
+                let result = connection.sock.write_all(bytes).map_err(|e| RedisError::from(e));
+                match result {
+                    Err(e) => {
+                        if e.is_broken_pipe() {
+                            connection.open = false;
+                        }
+                        Err(e)
+                    }
+                    Ok(_) => Ok(Value::Okay)
+                }
+            },
+        }
     }
 
     pub fn read_response(&mut self) -> RedisResult<Value> {
         let result = Parser::new(match *self {
-                ActualConnection::Tcp(ref mut reader) => reader as &mut Read,
+                ActualConnection::Tcp(TcpConnection{ ref mut reader, .. }) => reader as &mut Read,
                 #[cfg(any(feature="with-unix-sockets", feature="with-system-unix-sockets"))]
-            ActualConnection::Unix(ref mut sock) => &mut *sock as &mut Read,
+            ActualConnection::Unix(UnixConnection { ref mut sock, .. }) => sock as &mut Read,
```

Comment:
While you're at it, could you align this line with the `Tcp` one above?

Reply:
Done, are you using rustfmt in this project? My version (0.6.0) created a pretty ugly diff when I used it.

Reply:
I think it was used once and then manually corrected some stuff. It's not used in any automated way though.